### PR TITLE
*: add `analyze table` for mysql DB

### PR DESCRIPTION
**Pull request by bb7133**
When finish loading data, try to do `ANALYZE TABLE` for those DBs that support table analysis.

### CODE REVIEW

File: db/mysql/db.go [hunk 1]
```diff
@@ -325,6 +325,11 @@ func (db *mysqlDB) Delete(ctx context.Context, table string, key string) error {
 	return db.execQuery(ctx, query, key)
 }
 
+func (db *mysqlDB) Analyze(ctx context.Context, table string) error {
+	_, err := db.db.Exec(fmt.Sprintf(`ANALYZE TABLE %s`, table))
```
Comment (zhangjinpeng1987) [inline db/mysql/db.go]: Does it depends on the `db` has executed `use db` or the `db` is created with database?

File: pkg/client/client.go [hunk 2]
```diff
@@ -216,6 +216,12 @@ func (c *Client) Run(ctx context.Context) {
 	}
 
 	wg.Wait()
+	if !c.p.GetBool(prop.DoTransactions, true) {
+		// when loading is finished, try to analyze table if possible.
+		if analyzeDB, ok := c.db.(ycsb.AnalyzeDB); ok {
```
Comment (siddontang) [inline pkg/client/client.go]: If I remember correctly, db here is DBWrapper, so we can call Analyze directly, maybe.

File: pkg/client/dbwrapper.go [hunk 3]
```diff
@@ -165,3 +165,15 @@ func (db DbWrapper) BatchDelete(ctx context.Context, table string, keys []string
 	}
 	return nil
 }
+
+func (db DbWrapper) Analyze(ctx context.Context, table string) (err error) {
+	analyzeDB, ok := db.DB.(ycsb.AnalyzeDB)
+	if ok {
+		start := time.Now()
```
Comment (siddontang) [inline pkg/client/dbwrapper.go]: I don't think we need to measure this operation. If we want ,we can use ANALYZE directly, not `ANALYZE TABLE`

File: db/mysql/db.go (hunk 1 above)
Comment (siddontang) [reply db/mysql/db.go]: we create the Go Database with the specific database before, so no need here.

File: pkg/client/dbwrapper.go [hunk 4]
```diff
@@ -165,3 +165,15 @@ func (db DbWrapper) BatchDelete(ctx context.Context, table string, keys []string
 	}
 	return nil
 }
+
+func (db DbWrapper) Analyze(ctx context.Context, table string) (err error) {
```
Comment (siddontang) [inline pkg/client/dbwrapper.go]: remove err here.

File: pkg/client/client.go (hunk 2 above)
Comment (bb7133) [reply pkg/client/client.go]: `c.db` is of the `ycsb.DB` interface, so type assertion is required for `DBWrapper`:

https://github.com/pingcap/go-ycsb/blob/65aa46406f05ab9126991eec855b476c608985d6/pkg/client/client.go#L158

File: pkg/client/dbwrapper.go (hunk 3 above)
Comment (bb7133) [reply pkg/client/dbwrapper.go]: Okay

File: pkg/client/dbwrapper.go (hunk 4 above)
Comment (bb7133) [reply pkg/client/dbwrapper.go]: Okay

**Comment by bb7133** [comment]
Comments addressed, PTAL @siddontang @zhangjinpeng1987 , thanks!

File: pkg/client/dbwrapper.go [hunk 5]
```diff
@@ -165,3 +165,10 @@ func (db DbWrapper) BatchDelete(ctx context.Context, table string, keys []string
 	}
 	return nil
 }
+
+func (db DbWrapper) Analyze(ctx context.Context, table string) error {
```
Comment (zhangjinpeng1987) [inline pkg/client/dbwrapper.go]: `(db DbWrapper)` or `(db DbWrapper)`?

**Pull request merged**